server/common/src/main/java/org/eclipse/lsp/cobol/common/utils/RangeUtils.java

```diff
@@ -39,20 +39,31 @@ public static Optional<Node> findNodeByPosition(Node node, String uri, Position
     Node candidate = null;
     for (Node child : node.getChildren()) {
       if (isFromCopybook(uri, child)) {
-        candidate = child;
+        candidate = setCandidate(uri, child, candidate);
       }
 
       if (isInside(uri, position, child.getLocality())) {
-        candidate = child;
+        candidate = setCandidate(uri, child, candidate);
       }
       Optional<Node> nodeByPosition = findNodeByPosition(child, uri, position);
       if (nodeByPosition.isPresent()) {
-        return nodeByPosition;
+        candidate = nodeByPosition.get();
```

Comment:
added implementation looks in efficient, but the current node structure doesn't place the child nodes under CopyNode,
and we can't safely guarantee that found node is most appropriately matched until we scan the entire structure.